### ISSUE TITLE
Refactor C2 basis

### ISSUE THREAD

**Issue by orkolorko** [JuliaDynamics/RigorousInvariantMeasures.jl#146]
Due to the many refactors in interfaces, the C2 basis needs refactoring 
(also, it may be worth to implement the code for Linear Response as in [[1]](#1))

## References
<a id="1">[1]</a> 
Bahsoun, Galatolo, Nisoli, Niu - A rigorous computational approach to linear response 
DOI 10.1088/1361-6544/aa9a88

**Comment by fph** [comment]
True. I admit I don't know exactly what it does right now, and there is little documentation and testing so I am afraid to touch it. Is it even functional at the moment?

Maybe this could use a variant of `preimages_with_derivatives`, something like `preimages_derivatives_and_second_derivatives`. But maybe it is worth changing the assembler before; we had in our plans also to simplify the `dual+iterate` structure since there is a lot of cruft there that makes the code more complicated without adding much generality. I suggest to get that done before touching C2 and Chebyshev #147 .
